AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts

```diff
@@ -496,9 +531,47 @@ export class AddParticipantComponent extends BookingBaseComponent implements OnI
         return this.showDetails && this.searchEmail ? this.searchEmail.validateEmail() : true;
     }
 
+    validateJudgeAndJohMembers(): boolean {
+        if (this.hearing?.participants.length) {
+            const judge = this.hearing.participants.find(x => x.is_judge);
+            return this.searchEmail.email !== judge?.username;
+        }
+        return true;
+    }
+
+    validateJudiciaryEmailAndRole() {
+        if (this.searchEmail && this.searchEmail.email.length) {
+            this.searchService.searchJudiciaryEntries(this.searchEmail.email).subscribe(x => {
+                this.errorJudiciaryAccount = false;
+                if (x && x.length) {
+                    if (this.role.value !== 'Panel Member' && this.role.value !== 'Winger') {
```

Comment:
Both branches of this `if` call the same func. Is this right?

Reply:
Also the nesting looks a bit odd

Reply:
I _think_ I see what's happening, but perhaps we can make it a bit more readable.